Mathlib/MeasureTheory/MeasurableSpace/Basic.lean

```diff
@@ -1133,6 +1134,26 @@ namespace MeasurableSpace
   rintro _ ⟨u, -, rfl⟩
   exact (show MeasurableSet s from GenerateMeasurable.basic _ <| mem_singleton s).mem trivial
 
+theorem measurableSet_generateFrom_singleton_iff {s t : Set α} :
+    MeasurableSet[generateFrom {s}] t ↔ t = ∅ ∨ t = s ∨ t = sᶜ ∨ t = Set.univ := by
```

Comment:
I don't think we want this lemma. Wherever you want to use it, you should be using `generateFrom_singleton` directly.

This is of the same kind as people on Zulip asking how to show that all elements of some subgroup are of a certain form, rather than using the induction principles.

I too wanted it originally, and then I proved `generateFrom_singleton` and my problem was solved.

Reply:
If I correctly recall, my actual application for this lemma was showing that the cardinality of this set is 4, which appeared as a special case in a more general formula for the cardinality of the generated sigma algebra. An induction principle can't be directly used to prove that.

Reply:
Can we see the application?

Reply:
It's in a draft PR I have, I'll link it in an hour or two when I'm on my computer.